### ISSUE TITLE
[FEATURE_REQUEST] Clone persona

### ISSUE THREAD

**Issue by doomiiiii** [SillyTavern/SillyTavern#2558]
### Have you searched for similar requests?

No

### Describe the solution you'd like

Some character cards occasionally require specific scenarios that make using personas difficult unless you go and either modify it, at which point you'll need to undo it once no longer chatting with that specific bot, or create a brand new persona which is a bit of a pain
a simple solution to that would just be a button to clone personas

### Priority

Low (Nice-to-have)

### Are you willing to test this on staging/unstable branch if this is implemented?

Yes

**Comment by Wolfsblvt** [comment]
Cloning button isn't too bad as an idea. Make sensse to add.

Until then, can I suggest an alternative though?
If you are using a persona with a given name and image, but the content is shifting/changing, couldn't you just utilize WI and turn on/off the entries with the relevant persona description? Much more flexible this way. Your persona card can just be empty.

**Comment by Succubyss** [comment]
Honestly what I'd personally love is importing/exporting personas individually. You can only do them as a batch for some reason.

A way to export them as pngs with metadata in some form would be great too.

**Comment by Wolfsblvt** [comment]
That's the thing. Exporting/Importing isn't really easy, as there is no default format specified for personas. How would you do that? You cannot embed the metadata into the PNG, because it's not *really* a character.

Hacky workaround of pretending it's a character with just the description field filled is quite ugly. How would you know you can't/shouldn't import it as a character?

**Comment by Cohee1207** [comment]
Personas were not meant to be shareable, since they abstractly represent _you_ as a user. If a char.card makes any assumptions on what the user is as an entity, then it should either override the persona description (with optional {{original}} tag) or ignore it.

**Comment by cloak1505** [comment]
> That's the thing. Exporting/Importing isn't really easy, as there is no default format specified for personas. How would you do that? You cannot embed the metadata into the PNG, because it's not _really_ a character.

Literally the same way a character card json is embedded, except call it Persona instead of Chara then attach a base64 encoded string of `{"name":"Pisica","description":"{{user}} is a 28-year-old Romanian catgirl.","position":0}`. There's nothing special about a character card.

![png metadata](https://github.com/user-attachments/assets/1a12c482-a87d-4579-a5e4-14832ac8462d)

> How would you know you can't/shouldn't import it as a character?

Non-issue with the "Persona" metadata. Other than failing to import as a character, you'd probably download them to a persona folder, and whatever webpage hosting it (e.g. rentry) shouldn't dump it into a pile of character cards without saying it's a persona card.

**Comment by Wolfsblvt** [comment]
It is an issue. Because character PNG data is somewhat standardized. Persona is not standardized at all. It wouldn't make much sense to allow a general export in a format like PNG, is ST is the only app who could "understand" that.
Besides, this is no small change, the backend needs to be changed on how it writes/reads from PNG too.

Just feel like this is quite out of scope for what personas are supposed to be.

**Comment by cloak1505** [comment]
"I won't do it until other people do it." How did everything become "standard" before they were "standard"? Even standards aren't universally standard. Why does RIsu get to make up their own "v3 spec" that don't have their new fields supported by other frontends? 

You're overthinking things. When there's no standard for something at all because it hasn't been done, something simply rises and becomes de facto standard. Persona isn't something crazy or asking for *new* new software. The ability to read and write metadata from PNG exists. The concept of base64'ing a JSON exists. What doesn't exist is the exact setup of "hey, read key named X and use value Y here (and so on)."

The batch export itself is one bigger JSON. The first step would be to ignore PNG for now and implement individual persona JSON import/export. Slapping a metadata field onto PNG comes second. I understand that reworking ST's base functions around persona isn't as trivial as flipping a switch and will take some work, but the PNG itself isn't the biggest obstacle.

You don't need to care about the idea, but I don't see a need to actively vote *against* it. If no coders care enough to start, it simply doesn't happen.

**Comment by Wolfsblvt** [comment]
Wasn't meant to be that dismissive. If I came off that way, I apologize.

Export/import with just JSON sounds reasonable and doable.
I was just voicing my concern with building a new different system to embed data into PNGs. Not my decision anyway in the end.

If you are interested to see if someone picks this up, might be a good idea to open a separate issue, for import/export. As this one will get closed once the Dupe functionality is merged.

**Comment by Succubyss** [comment]
Exporting as PNGs was meant more as an aside; since, that way, you wouldn't have to reimport the persona avatar as well.

**Comment by Cohee1207** [comment]
The requested feature has been released. Please move other questions into a separate issue.
